src/OrchardCore.Modules/OrchardCore.Media/Shortcodes/ImageShortcodeProvider.cs

```diff
@@ -95,10 +98,20 @@ public ValueTask<string> EvaluateAsync(string identifier, Arguments arguments, s
                     queryStringParams.Add("rmode", mode);
                 }
 
+                if (className != null)
+                {
+                    className = "class=\"" + className + "\" ";
```

Comment:
Shouldn't it be HTML encoded?

Reply:
But then would need to check "-" is not encoded by the default encoder.

Reply:
On Line 115, content =  _htmlSanitizerService.Sanitize(content)  will encode the className

Reply:
Awesome, missed it, thanks.